validator/app.py

```diff
@@ -507,19 +563,72 @@ def fetch_book(vuid):
     data = df_domain[df_domain["vuid"] == vuid][["book_name", "vuid"]].rename(
         {"book_name": "name"}, axis=1
     )
-    data["vocabularies"] = [["domain", "innovation"]] * len(data)
-    data_json = json.loads(data.to_json(orient="records"))
-    return jsonify(data_json[0])
+    if data.empty:
+        _validate_vuid(vuid)
+        raise InvalidUsage("No such book", status_code=404)
+
+    data["vocabularies"] = [["domain", "innovation", "questions"]] * len(data)
+    page_list = (
+        df_innovation[df_innovation["cvuid"].str.startswith(vuid)]
+        .cvuid.str.split(":", expand=True)[1]
+        .tolist()
+    )
+    data_json = json.loads(data.to_json(orient="records"))[0]
```

Comment:
Maybe https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_dict.html would give you what you need?

Or if not, a method that abstracts the to_json/loads step might be useful.

Reply:
removed these in favor of to_dict